go/tools/go-upgrade/go_upgrade.go

```diff
@@ -0,0 +1,453 @@
+	"golang.org/x/exp/slices"
+)
+
+// The tool go_upgrade allows us to automate some tasks required
+// to bump the version of Golang used throughout our codebase.
+//
+// It comes with four commands:
+// 		- Default: `./go/tools/go-upgrade/go_upgrade.go`
+//
+// 		  This will simply bump the version across the codebase.
+// 		  The latest available version of Golang will be fetched
+// 		  and used instead of the old version.
+//
+// 		  By default, we do allow major Golang version upgrade such
+// 		  as 1.20 to 1.21 but this can be overridden using the
+// 		  allow-major-upgrade CLI flag. Usually, we only allow such
+// 		  upgrade on the main branch of the repository.
+//
+// 		  In CI, particularly, we do not want to modify the workflow
+// 		  files before automatically creating a Pull Request to
+// 		  avoid permission issues. The rewrite of workflow files can
+// 		  be disabled using the no-workflow-update CLI flag.
+//
+// 		  Moreover, this command automatically bumps the bootstrap
+// 		  version of our codebase. If we are on the main branch, we
+// 		  want to use the CLI flag `main` to remember to increment
+// 		  the bootstrap version by 1 instead of 0.1.
+//
+//
+// 		- Get Current Go Version: `./go/tools/go-upgrade/go_upgrade.go get_go_version`
+//
+// 		  This command outputs the currently used Golang version of
+// 		  our codebase.
+//
+//
+// 		- Get Current bootstrap Version: `./go/tools/go-upgrade/go_upgrade.go get_bootstrap_version`
+//
+// 		  This command outputs the currently used bootstrap version
+// 		  of our codebase.
+//
+//
+// 		- Update Workflows: `./go/tools/go-upgrade/go_upgrade.go update_workflows`
+//
+// 		  This command is used after the auto Pull Request has been
+// 		  created. One of the reviewer must go and use this command
+// 		  to update the workflows files with the newer Golang version.
+// 		  It takes two arguments `go-from` and `go-to` to identify the
+// 		  old Golang version (go-from) and the new one (go-to).
+
+const (
+	goDevAPI = "https://go.dev/dl/?mode=json"
+)
+
+type latestGolangRelease struct {
+	Version string `json:"version"`
+	Stable  bool   `json:"stable"`
+}
+
+func main() {
+	noWorkflowUpdate := flag.Bool("no-workflow-update", false, "Whether or not the workflow files should be updated. Useful when using this script to auto-create PRs.")
+	allowMajorUpgrade := flag.Bool("allow-major-upgrade", false, "Defines if Golang major version upgrade are allowed.")
+	isMainBranch := flag.Bool("main", false, "Defines if the current branch is the main branch.")
+	goFrom := flag.String("go-from", "", "The original Golang version we start with.")
+	goTo := flag.String("go-to", "", "The Golang version we want to upgrade to.")
+	flag.Parse()
+
+	switch {
+	case slices.Contains(os.Args, "get_go_version"):
+		currentVersion, err := currentGolangVersion()
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println(currentVersion.String())
+	case slices.Contains(os.Args, "get_bootstrap_version"):
```

Comment:
Oh, nice! First time I'm seeing this new ext package. That being said, IMO it's better to require the action be `get` and the first arg be `bootstrap-version`.  The current code would allow you to specify all sorts of things and not tell you that your usage is incorrect (and what the correct usage is) and make you think it's doing some things it's not.